llm-demo/infer_server.py

```diff
@@ -0,0 +1,802 @@
+"""
+
+import os
+
+os.getcwd()
+
+import time
+import warnings
+from argparse import ArgumentParser, ArgumentTypeError, Namespace
+from contextlib import nullcontext
+from typing import Any, Dict, List, Union
+
+import torch
+
+from transformers import (
+    AutoConfig,
+    AutoModelForCausalLM,
+    LlamaForCausalLM,
+    AutoTokenizer,
+    PreTrainedTokenizer,
+    PreTrainedTokenizerFast,
+    StoppingCriteria,
+    StoppingCriteriaList,
+    TextIteratorStreamer,
+    TextStreamer,
+)
+from termcolor import colored
+from peft import PeftModel
+from threading import Thread
+import intel_extension_for_pytorch as ipex
+
+Tokenizer = Union[PreTrainedTokenizer, PreTrainedTokenizerFast]
+
+import os
+import gradio as gr
+import time
+import socket
+import time
+
+print("==" * 20)
+hostname = socket.gethostname()
+IPAddr = socket.gethostbyname(hostname)
+print("IP Address is:" + IPAddr)
+
+ASSISTANT_COLOR = "light_cyan"
+ASSISTANT_BACKGROUND = None
+DEBUG = True
+
+
+def clear_bot():
+    return None, "", ""
+
+
+def log_feedback(feedback, ratings):
+    return "", 5
+
+
+title = """
+# <center> Meet Warren, your financial assistant </center>
+### <center> This model is based on Llama-2-chat-7b 🦙 and was fine-tuned on a finance dataset </center>
+"""
+
+
+md = """
+## <center>Developed with FastRAG</center>
+"""
+
+legal = """
+## Notices & Disclaimers:
+Performance varies by use, configuration and other factors. Learn more at www.Intel.com/PerformanceIndex. Performance results are based on testing as of dates shown in configurations and may not reflect all publicly available updates. See backup for configuration details. No product or component can be absolutely secure. Your costs and results may vary. Intel technologies may require enabled hardware, software or service activation. © Intel Corporation. Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries. Other names and brands may be claimed as the property of others.
+"""
+
+details = """
+## Software Configuration Details:
+<br>
+To learn more about the Intel software and tools used in this demo, visit developer.intel.com/ai
+"""
+
+hw = """
+## Hardware Details:
+Intel® Xeon® Platinum 8488C @2.4GHz, Cores per Socket: 4, Threads per Core: 2. Running on AWS m7i.2xlarge EC2 instance.
```

Comment:
Probably we shouldn't have this text there at all, or add functionality to report this properly. This text doesn't seems to be used anywhere (at least explicitly).

Reply:
Check other text fragments - do we really need them, or can drop

Reply:
Dropped the unnecessary ones. Only left title in since we use that